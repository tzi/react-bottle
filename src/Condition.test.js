import React from 'react';
import { Condition, Then, Else } from './Condition.component';
import renderer from 'react-test-renderer';

it('renders Then component', () => {
  const tree = renderer
    .create(
      <Condition if={true}>
        <Then>
          Then
        </Then>
        <Else if={true}>
          Else if
        </Else>
        <Else>
          Else
        </Else>
      </Condition>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Else component', () => {
  const tree = renderer
    .create(
      <Condition if={false}>
        <Then>
          Then
        </Then>
        <Else if={false}>
          Else if
        </Else>
        <Else>
          Else
        </Else>
      </Condition>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Else if component', () => {
  const tree = renderer
    .create(
      <Condition if={false}>
        <Then>
          Then
        </Then>
        <Else if={true}>
          Else if
        </Else>
        <Else>
          Else
        </Else>
      </Condition>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders via a render prop', () => {
  const tree = renderer
    .create(
      <Condition if={true}>
        <Then render={() => 'Then via render prop'}>
          This text would be override!
        </Then>
      </Condition>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
