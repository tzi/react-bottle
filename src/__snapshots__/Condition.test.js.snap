// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Else component 1`] = `"Else"`;

exports[`renders Else if component 1`] = `"Else if"`;

exports[`renders Then component 1`] = `"Then"`;

exports[`renders via a render prop 1`] = `"Then via render prop"`;
